tensorflow/python/summary/writer/event_file_writer.py

```diff
@@ -108,7 +112,9 @@ def close(self):
 
     Call this method when you do not need the summary writer anymore.
     """
+    self._worker.stop()
     self.flush()
+    self._worker.join()
```

Comment:
@mrry , I've tested, before queue get, I'll check queue size, so in my approach, the daemon thread will never be blocked by queue get, but in a busy waiting state(execute while True and continue again and again).

sentinel element is a better approach.